collector/metrics/metric_collector.go

```diff
@@ -124,10 +123,10 @@ func getReplicaAddrs() ([]string, error) {
 }
 
 // Register all metrics.
-func initMetrics() {
+func initMetrics(dataSource int) {
```

Comment:
How about making both the initMetrics function (some other functions are the same) and dataSource as members of MetricCollector, so the dataSource would not appear everywhere.


The global variables are the same.
var GaugeMetricsMap map[string]prometheus.GaugeVec
var CounterMetricsMap map[string]prometheus.CounterVec
var SummaryMetricsMap map[string]prometheus.Summary
var RoleByDataSource map[int]string
var TableNameByID map[string]string

Reply:
You can do the refactor in next patches, this PR looks good to me.

Reply:
OK，i will try.